--- lrp.py	2024-06-01 17:01:16.748442800 +0200
+++ lrp_modified.py	2024-06-01 17:08:18.874156300 +0200
@@ -1,5 +1,13 @@
 #!/usr/bin/env python3
 
+"""
+Modification of Captum's LRP to work in 3D.
+Rules have been added for nn.Conv3d, nn.BatchNorm3d and nn.LeakyReLU.
+
+Author: Captum
+source: https://github.com/pytorch/captum/blob/master/captum/attr/_core/lrp.py
+"""
+
 import typing
 from collections import defaultdict
 from typing import Any, cast, List, Tuple, Union
@@ -413,11 +421,14 @@
     nn.MaxPool2d: EpsilonRule,
     nn.MaxPool3d: EpsilonRule,
     nn.Conv2d: EpsilonRule,
+    nn.Conv3d: EpsilonRule,  # Added
     nn.AvgPool2d: EpsilonRule,
     nn.AdaptiveAvgPool2d: EpsilonRule,
     nn.Linear: EpsilonRule,
     nn.BatchNorm2d: EpsilonRule,
+    nn.BatchNorm3d: EpsilonRule,  # Added
     Addition_Module: EpsilonRule,
 }
 
-SUPPORTED_NON_LINEAR_LAYERS = [nn.ReLU, nn.Dropout, nn.Tanh]
\ No newline at end of file
+# Added nn.LeakyReLU
+SUPPORTED_NON_LINEAR_LAYERS = [nn.ReLU, nn.Dropout, nn.Tanh, nn.LeakyReLU]
\ No newline at end of file
